packages/kbn-search-connectors/lib/update_filtering.ts

```diff
@@ -6,59 +6,24 @@
  * Side Public License, v 1.
  */
 
+import { Result } from '@elastic/elasticsearch/lib/api/types';
 import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
 
-import { CONNECTORS_INDEX } from '..';
 import { fetchConnectorById } from './fetch_connectors';
-import {
-  Connector,
-  FilteringRule,
-  FilteringRules,
-  FilteringValidationState,
-} from '../types/connectors';
+import { FilteringRules } from '../types/connectors';
 
 export const updateFiltering = async (
   client: ElasticsearchClient,
-  connectorId: string,
-  {
-    advancedSnippet,
-    filteringRules,
-  }: {
-    advancedSnippet: string;
-    filteringRules: FilteringRule[];
-  }
+  connectorId: string
 ): Promise<FilteringRules | undefined> => {
-  const now = new Date().toISOString();
-  const parsedAdvancedSnippet: Record<string, unknown> = advancedSnippet
-    ? JSON.parse(advancedSnippet)
-    : {};
-  const parsedFilteringRules = filteringRules.map((filteringRule) => ({
-    ...filteringRule,
-    created_at: filteringRule.created_at ? filteringRule.created_at : now,
-    updated_at: now,
-  }));
-  const connector = await fetchConnectorById(client, connectorId);
-  if (!connector) {
-    throw new Error(`Could not find connector with id ${connectorId}`);
-  }
-  const active: FilteringRules = {
-    advanced_snippet: {
-      created_at: connector.filtering[0].active.advanced_snippet.created_at || now,
-      updated_at: now,
-      value: parsedAdvancedSnippet,
-    },
-    rules: parsedFilteringRules,
-    validation: {
-      errors: [],
-      state: FilteringValidationState.VALID,
-    },
-  };
-
-  const result = await client.update<Connector>({
-    doc: { ...connector, filtering: [{ ...connector.filtering[0], active, draft: active }] },
-    id: connectorId,
-    index: CONNECTORS_INDEX,
+  const activateDraftFilteringResult = await client.transport.request<{ result: Result }>({
+    method: 'PUT',
+    path: `/_connector/${connectorId}/_filtering/activate`,
   });
 
-  return result.result === 'updated' ? active : undefined;
+  if (activateDraftFilteringResult.result === 'updated') {
+    const connector = await fetchConnectorById(client, connectorId);
+    return connector?.filtering[0].active;
```

Comment:
Maybe unlikely but if `filtering` is `[]` this would throw an error `Uncaught TypeError: Cannot read properties of undefined (reading 'active')`.

This will either return active value or undefined
```suggestion
    return connector?.filtering.?[0]?.active;
```

Reply:
> Maybe unlikely but if filtering is [] this would throw an error 

Yeah, connectors are guaranteed, according to the protocol, to have filtering always populated with at least 1 entry. If filtering is empty it indicates an invalid state (which should not happen in theory). In that case we should return `undefined` (which would indicate sth is wrong in UI) - so accepting your suggestion. 👍